### PR TITLE
"Upgrading" WordPress

### PR DESCRIPTION
**Pull request by javiercasares**
This page will show some ideas on how to upgrade an old WordPress to a new one.

### CODE REVIEW

File: upgrading.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+IMPORTANT: This is a very manual process, not a massive one. There will be probably losses and some updates will be needed after the upgrade.
+
+[tip]In case only a major version (X.X) is indicated, use its minor (X.X.x) latest version available.[/tip]
+
+## Upgrading from WordPress 0.7 - 3.6 (by migration)
+
+Goals:
+- WordPress: upgrade to WordPress 6.0
+- PHP: upgrade to PHP 7.4
+- SQL: upgrade to MySQL 8.0 / MariaDB 10.5
```
Comment (grooverdan) [inline upgrading.md:24]: The versions here look to recent (going by the pattern with later migrations). Its going to be trouble if these DB versions are used and the next step attempts to downgrade them.

File: upgrading.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+- Plugins: all
+- Themes: all
+
+These are the oldest versions of WordPress and the ones that have not been supported for years. In general, have to assume some losses, although not of the contents, but probably of some functionality on themes and plugins.
+
+Considering that the goal is to keep the contents and assuming the loss of the rest of the elements, there are some steps.
+
+As with any upgrade, the first thing to do is to make a backup copy. The best way to upgrade from WP < 3.6 is to perform a content migration. 
+
+1. Create a brand-new WordPress, withoud the database.
```
Comment (grooverdan) [inline upgrading.md:37]: typo: without.

File: upgrading.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+2. Copy the old WordPress files from the "/wp-content/uploads/" content to the new one.
+3. Create a new database with the old database information. The best way is using "mysqldump".
+4. Configure the wp-config.php with all the new data.
+5. Access the "/wp-admin/" page, and follow the upgrading process.
+
+With this way, WordPress will be able to maintain and update the contents in the database and be able to work with these contents in an updated version of WordPress.
+
+A WordPress with the default theme, and all the contents should now be available.
+
+A common problem found in these database restorations is the character encoding type. It is very likely that its not in UTF-8 (maybe in an ISO or ASCII "deprecated" format), so make sure that the character encoding is updated correctly.
```
Comment (grooverdan) [inline upgrading.md:47]: suggest: detail how to check the character encoding

File: upgrading.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+Deactivate all left active plugins.
+
+Now, WordPress will have:
+-	Core: any version (between WordPress 3.7 and WordPress 4.0)
+-	Themes: Twenty Ten is active, and the main theme is deactivated.
+-	Plugins: all plugins that should be active are deactivated.
+
+At this point, overwrite the WordPress Core with [WordPress 4.1](https://wordpress.org/wordpress-4.1.zip), available in the [release list](https://wordpress.org/download/releases/). Install WordPress 4.1 major version or, if available and recommended, the latest 4.1.x minor version.
+
+Upgrade the systems up to PHP 5.6.20+ and MySQL 5.6.x or MariaDB 10.0.x. Please, do not install the newest major version.
```
Comment (grooverdan) [inline upgrading.md:84]: is there ambiguity between 5.6.20+ and not later version? alternate: "5.6.X (for X >= 20)", "5.6 (recommend 5.6.20+)"

File: upgrading.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+- WordPress: upgrade to WordPress 5.3
+- PHP: upgrade to PHP 7.4
+- SQL: maintain or upgrade to MySQL 8.0 / MariaDB 10.3
+
+Losses:
+- Content: none
+- Plugins: probably no
+- Themes: probably no
+
+_If you don't have PHP 7.2 configured yet, do it. Chances are that everything will still work normally._
```
Comment (grooverdan) [inline upgrading.md:147]: 7.2 or 7.4?

File: upgrading.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+WordPress 4.9 was the last version with the Classic Editor, so, a lot of people, afraid of the new editor, stopped updating WordPress. WordPress 5.0+ is fully compatible with the Classic Editor content, so it can be upgraded without losing any content.
+
+Also, when WordPress 4.9 was released, PHP 7.0+ was very stablished and 5 WordPress version had support. Upgrading from PHP 5.6 to PHP 7.0+ is very stable.
```
Comment (grooverdan) [inline upgrading.md:151]: 5 WordPress -> WordPress 5.0+

File: upgrading.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+From WordPress 4.9, you can continue with the manual update process, or start using [WP-CLI](https://wp-cli.org/), the tool to run WordPress commands directly via console, something that can easy the process.
```
Comment (grooverdan) [inline upgrading.md:153]: grammar: easy -> ease

File: upgrading.md [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+As with any upgrade, the first thing to do is to make a backup copy.
+
+Remove all themes that are not active, leaving only the main theme. If there is a child theme active, please, maintain the child and parent.
+
+Install and activate the [Twenty Ten](https://wordpress.org/themes/twentyten/) theme and activate it. This theme works in all sites since WordPress 3.7.
+
+In the same way, delete all deactivated plugins (and, therefore, not working).
```
Comment (grooverdan) [inline upgrading.md:161]: The "and, therefore, not working" reads a bit disjointed.

**Review by grooverdan — COMMENTED** [review]
review from me, a DB person amd not a WordPress expert.

File: upgrading.md (hunk 3 above)
Comment (kittenkamala) [reply upgrading.md:47]: Recommending that we update this wording to something like: 
"Character Encoding type commonly presents technical hiccups when restoring a database. It is possible that the backup data not encoded in UTF-8 (and instead may be in an ISO or ASCII "deprecated" format). Make sure that the character encoding is updated correctly upon restoring a database! [More information on converting Character Sets in a WordPress database can be found here.](https://codex.wordpress.org/Converting_Database_Character_Sets) "

File: upgrading.md (hunk 4 above)
Comment (kittenkamala) [reply upgrading.md:84]: > is there ambiguity between 5.6.20+ and a say a 7.0+ version? alternate: "5.6.X (for X >= 20)", "5.6 (recommend 5.6.20+)"

There are some incompatibilities between 5.6 and 7: https://www.php.net/manual/en/migration70.incompatible.php

Comment (javiercasares) [reply upgrading.md:84]: All the versions presented here are related to what was compatible with each WordPress version at the time. The idea is having the best version for each product.

**Comment by javiercasares** [comment]
First version prepared to be launched.

**Review by Crixu — APPROVED** [review]
Do we already need to add WP 6.1.1?

**Comment by javiercasares** [comment]
> Do we already need to add WP 6.1.1?

It has only major versions

**Pull request merged**